src/rust/engine/src/externs/address.rs

```diff
@@ -5,41 +5,53 @@ use pyo3::create_exception;
 use pyo3::exceptions::PyException;
 use pyo3::prelude::*;
 
-use address::parse_address;
-
 create_exception!(native_engine, AddressParseException, PyException);
 
 pub fn register(py: Python, m: &PyModule) -> PyResult<()> {
   m.add(
     "AddressParseException",
     py.get_type::<AddressParseException>(),
   )?;
-  m.add_function(wrap_pyfunction!(address_parse, m)?)?;
+  m.add_function(wrap_pyfunction!(address_spec_parse, m)?)?;
   Ok(())
 }
 
-// TODO: If more of `pants.build_graph.address.AddressInput` is ported to Rust, it might be worth
-// moving the definition into the `address` crate. But for now, this is a tuple.
+/// 1. a path component
+/// 2. a target component
+/// 3. a sequence of key/value parameters
+/// 4. a generated component
```

Comment:
These seem to be out of order with the field order in the type?

Reply:
Good catch.